src/Tool/WrapperInterface.php

```diff
@@ -63,7 +65,7 @@ public function hasValidIdentifier();
     /**
      * Get the object metadata.
      *
-     * @return ClassMetadata
+     * @return ClassMetadata&(OrmClassMetadata|OdmClassMetadata)
```

Comment:
Can we add a `@template TClassMetadata of ClassMetadata` to the interface? so `EntityWrapper` and `MongoDocumentWrapper` could specify their own `ClassMetadata`.

Not sure if it would solve other issues or maybe it would create more 😬